File: gaia-sdk-javascript/src/graphql/request/type/SkillProvisionBuildJob.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

export class SkillProvisionBuildJob extends Array<(_:VariableRegistry) => string> {
public _typeName = "SkillProvisionBuildJob";
    /**
     * Id of the tenant
     */
    public tenantId = () => { 
        this.push(_ => "tenantId")
    };

    /**
     * Reference to the skill provision for that build job
     */
    public provisionRef = () => { 
        this.push(_ => "provisionRef")
    };

    /**
     * Reference to the skill
     */
    public skillRef = () => { 
        this.push(_ => "skillRef")
    };

    /**
     * The name of the build job
     */
    public name = () => { 
        this.push(_ => "name")
    };

    /**
     * The current status of the build job
     */
    public status = () => { 
        this.push(_ => "status")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/SkillProvisionBuildJob.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class SkillProvisionBuildJob(list):

    """
    Id of the tenant
    """
    def tenant_id(self):
        self.append(lambda x: "tenantId")

    """
    Reference to the skill provision for that build job
    """
    def provision_ref(self):
        self.append(lambda x: "provisionRef")

    """
    Reference to the skill
    """
    def skill_ref(self):
        self.append(lambda x: "skillRef")

    """
    The name of the build job
    """
    def name(self):
        self.append(lambda x: "name")

    """
    The current status of the build job
    """
    def status(self):
        self.append(lambda x: "status")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
